Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5716,11 +5716,11 @@ void WebPageProxy::decidePolicyForNavigationAction(Ref<WebProcessProxy>&& proces
         && navigationActionData.navigationType != WebCore::NavigationType::BackForward;
 
     auto userInitiatedActivity = process->userInitiatedActivity(navigationActionData.userGestureTokenIdentifier);
-    auto navigationAction = API::NavigationAction::create(WTFMove(navigationActionData), sourceFrameInfo.get(), destinationFrameInfo.ptr(), std::nullopt, ResourceRequest(request), originalRequest.url(), shouldOpenAppLinks, WTFMove(userInitiatedActivity), mainFrameNavigation);
```

Comment:
Overall, this is an okay change, but taking one line of code and duplicating it into three places, which makes it harder to keep all the copies in sync.

Are we able to get to the `navigationData` from the `navigationAction` object after it's created?  Yes, we can through the `data()` method, so let's just keep this one line, and change the code below to stop using `navigationActionData` directly.

For example, `navigationActionData.` becomes `navigationAction.data().` on Line 5731:
```
    bool canHandleRequest = navigationActionData.canHandleRequest || m_urlSchemeHandlersByScheme.contains<StringViewHashTranslator>(request.url().protocol());
```

To this:
```
    bool canHandleRequest = navigationAction.data().canHandleRequest || m_urlSchemeHandlersByScheme.contains<StringViewHashTranslator>(request.url().protocol());
```

Line 5732 is the only other change I think.